src/Form/Expression/ConditionForm.php

```diff
@@ -111,9 +111,19 @@ public function form(array $form, FormStateInterface $form_state) {
   }
 
   /**
+   * {@inheritdoc}
+   */
+  public function validateForm(array $form, FormStateInterface $form_state) {
+    if ($form_state->get('condition')) {
+      $this->submitForm($form, $form_state);
```

Comment:
We should have a comment here why we call the submit handler here. Can be similar to the one we had in AddExpressionForm.